src/coursier/coursier.ts

```diff
@@ -0,0 +1,12 @@
+import { downloadCoursierIfRequired } from "./download-coursier";
+import { findCoursierOnPath } from "./path-check";
+
+export function getCoursierExecutable(extensionPath: string): Promise<string> {
```

Comment:
I tried your branch locally, and the extension dowloaded coursier, and ran the downloaded binary, and it just worked, with is great ... except that I have coursier installed :D 

I think you forgot to wire this function into the main flow

Reply:
ahahha, you're so right, I did not have time to test e2e before I left for my appointment

fixing this ASAP

Reply:
tested the latest commit: [deddb59](https://github.com/disneystreaming/vscode-smithy/pull/5/commits/deddb5960bb3d998da8e1b696c1b925372e3e53c)

and it worked correctly, (not downloading in my case because I had the coursier binary on my path)